services/src/main/java/org/fao/geonet/api/records/attachments/AttachmentsApi.java

```diff
@@ -241,47 +249,62 @@ public MetadataResource putResourceFromURL(
-                    BufferedImage image = ImageIO.read(file.getPath().toFile());
-                    BufferedImage resized = ImageUtil.resize(image, size);
-                    ByteArrayOutputStream output = new ByteArrayOutputStream();
-                    ImageIO.write(resized, "png", output);
-                    output.flush();
-                    byte[] imagesB = output.toByteArray();
-                    output.close();
-                    return new HttpEntity<>(imagesB, headers);
+
+        ApiUtils.canViewRecord(metadataUuid, request);
+
+        ServletOutputStream out = response.getOutputStream();
+
+        if (store instanceof FMEStore
+        || (store instanceof ResourceLoggerStore && ((ResourceLoggerStore) store).getDecoratedStore() instanceof FMEStore)) {
+            response.setHeader("Content-Disposition", "inline; filename=\"" + resourceId + "\"");
+            response.setHeader("Cache-Control", "no-cache");
+
+            store.streamResource(context, metadataUuid, resourceId, approved, out);
+        } else {
+            try (Store.ResourceHolder file = store.getResource(context, metadataUuid, resourceId, approved)) {
+                response.setHeader("Content-Disposition", "inline; filename=\"" + file.getMetadata().getFilename() + "\"");
+                response.setHeader("Cache-Control", "no-cache");
+
+                Path path = file.getPath();
+                String contentType = getFileContentType(path);
+                response.setHeader("Content-Type", contentType);
+
+                if (contentType.startsWith("image/") && size != null) {
+                    if (size >= MIN_IMAGE_SIZE && size <= MAX_IMAGE_SIZE) {
+                        BufferedImage image = ImageIO.read(path.toFile());
+                        BufferedImage resized = ImageUtil.resize(image, size);
+                        ByteArrayOutputStream output = new ByteArrayOutputStream();
+                        ImageIO.write(resized, "png", output);
+                        output.flush();
+                        byte[] imagesB = output.toByteArray();
+                        output.close();
+                        out.write(imagesB);
+                        out.flush();
+                        out.close();
+                    } else {
+                        throw new IllegalArgumentException(String.format(
+                            "Image can only be resized from %d to %d. You requested %d.",
+                            MIN_IMAGE_SIZE, MAX_IMAGE_SIZE, size));
+                    }
                 } else {
-                    throw new IllegalArgumentException(String.format(
-                        "Image can only be resized from %d to %d. You requested %d.",
-                        MIN_IMAGE_SIZE, MAX_IMAGE_SIZE, size));
+                    IOUtils.copy(Files.newInputStream(path, StandardOpenOption.READ), out);
+                    out.flush();
+                    out.close();
```

Comment:
Closing the HttpServletResponse's outputStream is not needed. The servlet container will do it.